### ISSUE TITLE
very very slow when open 2files with more than 2000 lines

### ISSUE THREAD

**Issue by mjheidari** [mattpass/ICEcoder#673]
when use multi file that have about 2K lines or more,it's very slow.for example when you type ,each character type in one second!!

**Comment by mattpass** [comment]
Yes, does seem to be quite a slowdown. I'll see what's causing that and resolve.

**Comment by mattpass** [comment]
OK, found the issue (I tested on a 8,000 line file) and saw a slowdown when typing. Running a JS profiler, I've found it's calling setMiniMap 0.1s after typing and that regenerates the minimap to thr right hand side. It also has this slowdown when changing tabs (the only 2 places setMiniMap is called).

Approaches to solve this could be: only allow setMiniMap to run in quiet times (ie when not typing) and/or detect what has changed and make those changes in the minimap also. Re switching tabs. keep a copy of the minimap for each tab and use that rather than regenerate.
